### ISSUE TITLE
Loop condition contains call to external function / method

### ISSUE THREAD

**Issue by stefandrissen** [Riverside-Software/sonar-openedge#826]
This rule should only flag when the TO part contains an external function / method since that is executed per iteration of the loop. When the from part contains an external function / method it should not be flagged:

````
function foo returns integer ():
   return 10.
end function.

def var ic as int no-undo.

do ic = 1 to foo(): // should be flagged
   // do something foo times
end.

do ic = foo() to 1 by -1: // should NOT be flagged
   // do something foo times
end.
````

**Comment by gquerret** [comment]
Confirmed.

**Comment by gquerret** [comment]
Fixed, will be included in 2.9
